gtfs-realtime/spec/en/reference.md

```diff
@@ -340,9 +354,89 @@ Identification information for the vehicle performing the trip.
 
 | _**Field Name**_ | _**Type**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|-------------------|-------------------|
-| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field |
+| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field. |
 | **label** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | User visible label, i.e., something that must be shown to the passenger to help identify the correct vehicle. |
 | **license_plate** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | The license plate of the vehicle. |
+| **carriage_descriptor** | [CarriageDescriptor](#message-carriagedescriptor) | repeated | An ordered list of carriage information. |
```

Comment:
I think I understand your idea. My objection was that it is probably not a feasible solution for cases where such train is modeled as a single trip (which can have only a single value of `direction_id`). It would require to slice the train journey to several trips arbitrarily just for the sake of export to GTFS. Even so, using `direction_id` for this purpose still feels wrong to me, as the train (even with the order of carriages reversed) still follows the same direction in the sense of trips.txt's `direction_id` semantics.

Take the [Copenhagen-Karlskrona](https://goo.gl/maps/X3AZg8mKuj82) route as an example:
`Copenhagen-Kristianstad-Karlskrona, direction_id=0`
`Karlskrona-Kristianstad-Copenhagen, direction_id=1`
and the order of carriages has got always reversed in Kristianstad.

And if `CarriageDescriptor[]` would be tight to a `TripDescriptor` only, then while the train would be in the Copenhagen-Kristianstad section, the feed would advertise misleading order of carriages for the rest of its journey after Kristianstad. That might be quite inconvenient for passengers boarding at stations in that section and checking the status of the train in advance.